linux/just_files/just_docker_functions.bsh

```diff
@@ -517,6 +528,39 @@ function docker_defaultify()
       extra_args=${#}
       ;;
 
+    docker-compose_debug) # Run docker buildx debug build against a \
+        # docker-compose service. $1 - service name (required) $2 - docker context, defaults to "."
+      # These are not a perfect parse? Won't handle yaml strings that aren't the same syntax as bash strings
+      local compose_config=$(docker-compose config | yarp)
+      # This shouldn't be blank, as if it's absent in the compose file, docker-compose will report "Dockerfile" still
+      local dockerfile=$(sed -nE 's|^services\.'"$1"'\.build\.dockerfile = "?(.*[^"])"?|\1|p' <<< "${compose_config}")
+      if [ -z "${dockerfile}" ]; then
+        echo "Dockerfile is empty, either you forgot to or specified the wrong service name" >&2
+        JUST_IGNORE_EXIT_CODES=1
+        return 1
+      fi
+      # This won't even handle spaces
+      local build_args=($(sed -nE 's|^services\.'"$1"'\.build\.args\.([^ ]*) = "?(.*[^"])"?|--build-arg \1=\2|p' <<< "${compose_config}"))
+
+      if [ -n "${2+set}" ]; then
+        if [ "${2}" = "*" ]; then
+          build_args+=(--no-cache)
+        else
+          build_args+=(--no-cache-filter "${2}")
+        fi
+      fi
+
+      # Requires buildx 0.12 or ?newer? (syntax may change, as this is an experimental feature)
```

Comment:
Should there be a buildx version check?

Reply:
Given the experimental nature of this, I chose not to. Once the API settles down and it's more predictable, yes

Reply:
perhaps a TODO add version check once out of the experimental stage